### DISCUSSION TITLE
GitHub Cache with Maven project in subdirectory

### DISCUSSION THREAD

**Discussion by akefirad** [community/community#26902] [Actions]
<p>I had the following configuration of my GH Action workflow of a simple Maven project.</p>
<pre><code class="lang-yaml">name: Java CI

on:
  pull_request:
    branches:
      - master

jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 11
        uses: actions/setup-java@v1
        with:
          java-version: 11
      - name: Cache Maven repository
        uses: actions/cache@v2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
      - name: Verify with Maven
        run: mvn -B -e -ff clean verify
</code></pre>
<p>Everything used to work, till I move the Maven project into a <code>code</code> subdirectory:</p>
<pre><code class="lang-yaml">name: Java CI

on:
  pull_request:
    paths:
      - code/**
    branches:
      - master

jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK 11
        uses: actions/setup-java@v1
        with:
          java-version: 11
      - name: Cache Maven repository
        uses: actions/cache@v2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
      - name: Verify with Maven
        run: mvn -B -e -ff clean verify --file code/pom.xml
</code></pre>
<p>Now in the log I see the cache step says <code>Cache not found for input keys: Linux-maven-xxxxxxx, Linux-maven-</code>.</p>
<p>Does anyone know why the cache is not working with the project in a subdirectory?<br>
Thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/akefirad">@akefirad</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>I followed your description and copied your code to my repo, however cache works fine after i move the project to subdirectory <code>code</code>. My workflow <a href="https://github.com/weide-zhou/packages-example-for-support/runs/1115468723?check_suite_focus=true#step:4:14" rel="noopener nofollow ugc">here</a>, and screenshot below:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/71b68952dd727b5264799f1c79cff3df26899876.png" data-download-href="/uploads/short-url/gdX2gPWWBfIMIDP39dQu5zlFzx4.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/52772761/181100533-fa707cac-1241-4fec-84a5-772f97869b45.png" alt="image" data-base62-sha1="gdX2gPWWBfIMIDP39dQu5zlFzx4" width="690" height="74" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/71b68952dd727b5264799f1c79cff3df26899876_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1255×136 11.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The <code>cache miss</code> typically appears in below situation:</p>
<ol>
<li>It’s first time to run the workflow, and never cached before.</li>
<li>The key or restore key doesn’t match.</li>
</ol>
<p>And GitHub will remove any cache entries that have not been accessed in over 7 days. Did you cache it recently?</p>
<p>Please kindly have a double check and if the issue persists, please share your repo/sample repo for further investigation.</p>
<p>Thanks</p>

**Comment by akefirad** [comment]
<p>I’m pretty sure I checked it multiple times and didn’t see the cache being used. But you’re right. Now it’s working. Not sure if it was really because of <span class="hashtag">#1</span> or <span class="hashtag">#2</span>. Sorry for the bother. Thanks.</p>
